### PR TITLE
feat(terraform): allow minimal naming convention

### PR DESCRIPTION
**Pull request by secustor**
This PR allows this kind of definitions of providers and reduces double handling of logic.  

```hcl
terraform {
  required_providers {
    docker = {
      source  = "docker"
      version = "2.7.2"
    }
  // translates to this
   docker2 = {
      source  = "registry.terraform.io/hashicorp/docker"
      version = "2.7.2"
    }
  }
  required_version = ">= 0.13"
}
```

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
This needs tests to get coverage back to 100 % 🙃

**Review by viceice — COMMENTED** [review]
This needs to be tested more, I'm unsure if the registry url changes are correct.

File: lib/manager/terraform/__fixtures__/1.tf [hunk 1]
```diff
@@ -178,6 +178,10 @@ terraform {
       source  = "terraform-providers/docker"
       version = "2.7.2"
     }
+    docker2 = {
+      source  = "docker"
```
Comment (viceice) [inline lib/manager/terraform/__fixtures__/1.tf:182]: Are you sure this is valid? As far as I know Terraform 0.13 throws an error on this.

Comment (secustor) [reply lib/manager/terraform/__fixtures__/1.tf:182]: The reasoning behind this is this documentation: https://www.terraform.io/docs/configuration/provider-requirements.html#source-addresses   
Specifically the note at the bottom.

I have tested it with this code: 

``` hcl 
terraform {
  required_providers {
    aws = {
      source  = "aws"
      version = "~> 3.0"
    }
    kubernetes = {
      source  = "kubernetes"
      version = "~> 1.0"
    }
  }
}
```
image

**Comment by secustor** [comment]
Can you please elaborate what kinds of test you want to have added?

The registry URLs are the part which I have referenced with `double handling of logic` this logic is already present in the datasource.   
It uses the hunt approach and tries first to lookup dependency in the registry and then, for backwards compatibility to Terraform 0.12, `releases.hashicorp.com`

**Comment by viceice** [comment]
Can you please create a sample repo with some pr to show it works as expected.

**Comment by secustor** [comment]
I have created a new testing repo with only the required_providers definition.
https://github.com/secustor/renovate_test

File: lib/manager/terraform/__fixtures__/1.tf (hunk 1 above)
Comment (viceice) [reply lib/manager/terraform/__fixtures__/1.tf:182]: ok, you've removed `docker` in your real test.

```yaml
terraform {
  required_providers {
    docker = {
      source  = "docker"
      version = "2.7.2"
    }
    helm = {
      source  = "helm"
      version = "1.3.2"
    }
    kubernetes = {
      source  = "kubernetes"
      version = "1.13.2"
    }
  }
  required_version = ">= 0.13"
}
```
image

so can you please update the test case to me more real world 🙃

Comment (secustor) [reply lib/manager/terraform/__fixtures__/1.tf]: I will do that!

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**